sklearn/metrics/pairwise.py

```diff
@@ -1886,6 +1886,15 @@ def pairwise_distances_chunked(
         yield D_chunk
 
 
+@validate_params(
+    {
+        "X": [np.ndarray],
+        "Y": [np.ndarray, None],
```

Comment:
The error is because the input can be some sparse matrices.

```suggestion
        "X": ["array-like"],
        "Y": ["array-like", None],
```

Reply:
Additionally, we need to update the docstring of `X` and `Y` to mention that we accept array-like and sparse matrix.

Reply:
This is more complex than that. Sparse matrices are only supported for scikit-learn metrics. So we should allow sparse matrix. However, there is a bug because we allow passing sparse matrix to `cdist` while a better error should have been raised earlier.

I fixed this issue by setting `accept_sparse=False` in the branch using `cdist` and adding a unit test to ensure we raise the proper error message. In this way, it should also fix the failing test observed with `OPTICS` which was a side effect of getting a non-informative error message.

Reply:
Uhm actually it was not a bug. This is the side effect of having to remove the `raise TypeError`. I will remove the entry from the changelog then.

Reply:
I revert the code such that we keep the previous one. I assume that we would prefer the change by adding `accept_sparse=False` but it is better to keep this PR as close as possible to the original code. We can always revisit to make some maintenance later.